ios/Sources/GutenbergKit/Sources/EditorLibrary.swift

```diff
@@ -0,0 +1,222 @@
+        rootDirectory.appendingPathComponent("index.html")
+    }
+}
+
+public struct LocalEditorManifest: Hashable {
+    public let rootDirectory: URL
+
+    init(_ url: URL) {
+        let hash = url.deletingPathExtension().lastPathComponent
+        self.init(hash: hash)
+    }
+
+    init(rootDirectory: URL) {
+        self.rootDirectory = rootDirectory
+    }
+
+    init(hash: String) {
+        self.rootDirectory = editorCacheRoot.appendingPathComponent("\(hash).editorbundle")
+    }
+
+    public var editorURL: URL {
+        rootDirectory.appendingPathComponent("index.html")
+    }
+}
+
+/// A mechanism to store editor assets on-disk to avoid downloading them every time
+///
+public actor EditorLibrary {
+
+    enum CachedAssetType {
+        case script
+        case style
+    }
+
+    struct CachedAsset {
+        let type: CachedAssetType
+        let source: String
+        let destination: String
+    }
+
+    public nonisolated var bundledManifest: LocalEditorManifest {
+        LocalEditorManifest(rootDirectory: Bundle.module.url(forResource: "Gutenberg", withExtension: nil)!)
+    }
+
+    public nonisolated var rootDirectory: URL {
+        editorCacheRoot
+    }
+
+    public typealias ProgressCallback = (Progress) -> Void
+
+    private let urlsession: URLSession
+
+    public init(urlSession: URLSession = .shared) {
+        self.urlsession = urlSession
+    }
+
+    /// Downloads the editor asset manifest from the given URL
+    ///
+    /// Optionally takes a progress callback for updating UI
+    /// ```swift
+    /// // While the user is waiting:
+    /// try await EditorAssetCache().buildManifest(for: manifest) { progress in
+    ///     print(progress.fractionCompleted)
+    /// }
+    ///
+    /// // In the background, prewarming the cache:
+    /// try await EditorAssetCache().buildManifest(for: manifest)
+    /// ```
+    ///
+    ///
+    @discardableResult
+    public func downloadManifest(from url: URL, progress callback: ProgressCallback?) async throws -> LocalEditorManifest {
+        let (data, _) = try await urlsession.data(from: url)
+        let manifest = try EditorManifest(data: data)
+
+        try await buildManifest(for: manifest, progress: callback)
+
+        return LocalEditorManifest(hash: manifest.hash)
+    }
+
+    func buildManifest(for manifest: EditorManifest, progress callback: ProgressCallback? = nil) async throws {
+        try FileManager.default.createDirectory(at: manifest.rootDirectory, withIntermediateDirectories: true)
+
+        let assets = try await withThrowingTaskGroup(of: CachedAsset.self) { group in
+            let progress = Progress(totalUnitCount: Int64(manifest.scripts.count + manifest.styles.count))
+            callback?(progress)
+
+            var assets: [CachedAsset] = []
+
+            for script in manifest.scripts {
+                group.addTask {
+                    let destination = manifest.rootDirectory.appendingPathComponent(script.path)
```

Comment:
Verify if this approach works with sites that are located under a subdirectory: `http://example.com/blog`